metrics/prometheus.go

```diff
@@ -147,6 +155,61 @@ func (o *prometheusMetrics) newHistogramMeter(name string, buckets []int64) Hist
 	}
 }
 
+func getIOLineValue(line string) (int64) {
+	fields := strings.Fields(line)
+	if len(fields) != 2 {
+		logger.Warn("this io file line is malformed", "err", line)
+		return 0
+	}
+	value, err := strconv.ParseInt(fields[1], 10, 64)
+	if err != nil {
+		logger.Warn("unable to parse int", "err", err)
+		return 0
+	}
+
+	return value
+}
+
+func getDiskIOData() (int64, int64, error) {
+	pid := os.Getpid()
+	ioFilePath := fmt.Sprintf("/proc/%d/io", pid)
+	file, err := os.Open(ioFilePath)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	// Parse the file line by line
+	scanner := bufio.NewScanner(file)
+	var reads, writes int64
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "syscr") {
+			reads = getIOLineValue(line)
+		} else if strings.HasPrefix(line, "syscw") {
+			writes = getIOLineValue(line)
+		}
+	}
+
+	return reads, writes, nil
+}
+
+func (o *prometheusMetrics) collectDiskIO(refresh time.Duration) {
+	for {
+		reads, writes, err := getDiskIOData()
+		if err != nil {
+			continue
```

Comment:
if `getDiskIOData` keeps returning an error this for loop will iterate without stop indefinitely

Reply:
Sure thing man, I missed this, thanks for pointing out (changed in the last commit)